autogpts/autogpt/autogpt/core/resource/model_providers/openai.py

```diff
@@ -406,83 +411,90 @@ async def create_chat_completion(
     ) -> ChatModelResponse[_T]:
         """Create a completion using the OpenAI API."""
 
-        completion_kwargs = self._get_completion_kwargs(model_name, functions, **kwargs)
-        tool_calls_compat_mode = functions and "tools" not in completion_kwargs
-        if "messages" in completion_kwargs:
-            model_prompt += completion_kwargs["messages"]
-            del completion_kwargs["messages"]
+        openai_messages, completion_kwargs = self._get_chat_completion_args(
+            model_prompt, model_name, functions, **kwargs
+        )
+        tool_calls_compat_mode = bool(functions and "tools" not in completion_kwargs)
```

Comment:
Yes, but I'd rather implement that at a higher level. I want to have a wrapper function that determines which model and/or which provider to use, so that it's possible to use multiple providers throughout the application.